extension-qr-code/src/org/eclipse/birt/extension/qrcode/util/SwingGraphicsUtil.java

```diff
@@ -0,0 +1,67 @@
+/*******************************************************************************
+ * Copyright (c) 2022 Henning von Bargen
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * https://www.eclipse.org/legal/epl-2.0/.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ *
+ * Contributors:
+ *  Henning von Bargen - initial API and implementation
+ *******************************************************************************/
+
+package org.eclipse.birt.extension.qrcode.util;
+
+import java.awt.image.BufferedImage;
+import java.util.HashMap;
+
+import com.google.zxing.BarcodeFormat;
+import com.google.zxing.EncodeHintType;
+import com.google.zxing.client.j2se.MatrixToImageConfig;
+import com.google.zxing.client.j2se.MatrixToImageWriter;
+import com.google.zxing.common.BitMatrix;
+import com.google.zxing.qrcode.QRCodeWriter;
+import com.google.zxing.qrcode.decoder.ErrorCorrectionLevel;
+
+/**
+ * SwingGraphicsUtil
+ */
+public class SwingGraphicsUtil {
+
+	public static BufferedImage createQRCodeImage(String text, int dotsWidth, int dotsHeight, String encoding,
```

Comment:
"create" method should never return `null`, it should throw exception instead if creation failed

Reply:
I considered changing this, but I didn't, because the code wouldn't be cleaner IMHO. 
Besides, returning null and more or less silently ignore many exceptions seems to be the traditional BIRT coding style.